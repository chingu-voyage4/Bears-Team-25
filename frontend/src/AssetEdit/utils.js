const getFileType = (file, availableTypes = []) => {
  if (
    !file ||
    !file.type ||
    (typeof availableTypes !== 'object' && !availableTypes.length)
  ) {
    return 'other';
  }

  let fileType = file.type.split('/')[0] || 'other';

  if (!availableTypes.includes(fileType)) {
    fileType = 'other';
  }

  return fileType;
};

const getMediaType = d64 => {
  if (!d64) {
    return '';
  }

  if (typeof d64 !== 'string') return 'other';

  const match = d64.match(/(:)(\w+)(\/)/);

  if (!match || match.length < 3) {
    return 'other';
  }

  return match[2];
};

const readFile = file =>
  new Promise(resolve => {
    if (!(file instanceof File)) {
      resolve('');
    }

    const reader = new FileReader();

    reader.addEventListener('load', () => {
      resolve(reader.result);
    });

    reader.readAsDataURL(file);
  });

const getLocalUrl = blob => URL.createObjectURL(blob);

const packageData = ({ _id, file, title, description }) => {
  const payload = new FormData();

  if (_id) {
    payload.append('_id', _id);
  }

  payload.append('title', title);
  payload.append('description', description);

  if (file) {
    payload.append('blob', file);
  }

  return payload;
};

export { getFileType, getMediaType, packageData, readFile, getLocalUrl };
