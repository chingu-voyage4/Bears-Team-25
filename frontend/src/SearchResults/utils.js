export const fixPath = (path = '') => {
  if (typeof path !== 'string') {
    return '';
  }

  return path.split('>').join(' > ');
};

export const getIndexes = (source, query) => {
  if (typeof source !== 'string' || typeof query !== 'string') {
    return [];
  }

  const result = [];

  for (let i = 0; i < source.length; i += 1) {
    if (source.substring(i, i + query.length).toLowerCase() === query) {
      result.push(i);
    }
  }

  return result;
};

export const stayPositive = (a, b) => {
  if (typeof a !== 'number' || typeof b !== 'number') {
    return 0;
  }

  return a - b < 0 ? 0 : a - b;
};
