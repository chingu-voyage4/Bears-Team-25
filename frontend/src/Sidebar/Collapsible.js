import React from 'react';
import PropTypes from 'prop-types';

import { Details, Summary } from './styled';

export default class Collapsible extends React.Component {
  static propTypes = {
    id: PropTypes.string.isRequired,
    children: PropTypes.oneOfType([PropTypes.object, PropTypes.array])
      .isRequired,
    title: PropTypes.string.isRequired,
    // eslint-disable-next-line react/no-unused-prop-types
    open: PropTypes.bool.isRequired,
    expanded: PropTypes.func.isRequired,
  };
  // eslint-disable-next-line
  state = {
    open: false,
  };
  static getDerivedStateFromProps(nextProps) {
    return { open: nextProps.open };
  }
  onClick = e => {
    e.stopPropagation();
    const { open } = this.state;
    this.props.expanded(this.props.id, !open);
    this.setState({ open: !open });
  };
  render() {
    const { title, children } = this.props;
    const { open } = this.state;
    const rightArrow = String.fromCharCode(9658);
    const downArrow = String.fromCharCode(9660);
    return (
      <Details>
        <Summary onClick={this.onClick}>
          {open ? downArrow : rightArrow} {title}
        </Summary>
        {open && children}
      </Details>
    );
  }
}
