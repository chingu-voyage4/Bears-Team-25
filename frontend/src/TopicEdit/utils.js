// eslint-disable-next-line
export const sortByOrder = arr => {
  const copy = [...arr];
  copy.sort((a, b) => a.order - b.order);
  return copy.map((n, i) => ({ ...n, order: i }));
};

export const normalizeChildren = arr => {
  const norm = arr.reduce((acc, b) => {
    if (!acc[b.parent]) {
      acc[b.parent] = [];
    }
    acc[b.parent].push(b);
    return acc;
  }, {});

  Object.keys(norm).forEach(key => {
    norm[key].sort((a, b) => a.order - b.order);
    norm[key] = norm[key].map((k, i) => ({ ...k, order: i }));
  });

  return norm;
};

export const prepareParents = arr => arr.map((a, i) => ({ ...a, order: i }));

export const prepareChildren = obj => {
  const result = Object.values(obj).reduce((acc, arr) => {
    acc.push(...arr.map((a, i) => ({ ...a, order: i })));
    return acc;
  }, []);
  result.sort((a, b) => a._id < b._id);
  return result;
};
