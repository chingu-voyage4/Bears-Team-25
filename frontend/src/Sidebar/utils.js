import React from 'react';

import Collapsible from './Collapsible';

import { LI, DrawerLink } from './styled';

export const checkMobile = (mobile, windowWidth) => {
  const BREAK_MOBILE = 900;

  if (
    (mobile && windowWidth > BREAK_MOBILE) ||
    (!mobile && windowWidth <= BREAK_MOBILE)
  ) {
    return !mobile;
  }

  return mobile;
};

/* eslint-disable spaced-comment */
/******************************************************************************

{
  HELP : {
    About this CMS : {
      About this CMS: "5ae27568463f344afc3ac09f"
    }
  }
  How to ... : {
    Create a New Article: "5ae27568463f344afc3ac0a0",
    Edit an Article: "5ae27568463f344afc3ac0a1"
  }
  Introduction : "5ae27568463f344afc3ac09e"
}

new tree structure
we need expanded flag and _id for topic/sub-topics, but not article leaf nodes

{
  HELP : {
    _id: string
    expanded: [true|false]
    About this CMS : {
      _id: string
      expanded: [true|false]
      About this CMS: "5ae27568463f344afc3ac09f"
    }
    How to ... : {
      _id: string
      expanded: [true|false]
      Create a New Article: "5ae27568463f344afc3ac0a0",
      Edit an Article: "5ae27568463f344afc3ac0a1"
    }
    Introduction : "5ae27568463f344afc3ac09e"
  }
  next-topic: {}
}

******************************************************************************/

export const getTree = articles => {
  const tree = {};
  articles.forEach(article => {
    if (!tree[article.topic.name]) {
      tree[article.topic.name] = {
        _id: article.topic._id,
        expanded: false,
      };
    }
    let topic = tree[article.topic.name];
    if (article.sub_topic) {
      const list = article.sub_topic.name.split('>');
      list.forEach(sub => {
        if (!topic[sub]) {
          topic[sub] = {
            _id: article.sub_topic._id,
            expanded: false,
          };
        }
        topic = topic[sub];
      });
    }
    topic[article.title] = article._id;
  });
  return tree;
};

let ndx = 0;
/* eslint-disable no-plusplus */
export const getChildren = (sub, path, onArticleSelect, onExpand) => {
  const keys = Object.keys(sub);
  return keys.reduce((acc, key) => {
    if (key === '_id' || key === 'expanded') return acc;
    const s = sub[key];
    if (typeof s === 'string') {
      // TODO: great idea but keeping it updated will be fun
      // const displayed = path.includes(key) ? ' >' : '';
      return acc.concat(
        <LI key={ndx++} onClick={onArticleSelect}>
          <DrawerLink to={`/cms/${s}`}>{`${key}`}</DrawerLink>
        </LI>
      );
    }
    const children = getChildren(sub[key], path, onArticleSelect, onExpand);

    const open = sub[key].expanded || path.includes(key);
    return acc.concat(
      <Collapsible
        key={ndx++}
        title={key}
        id={sub[key]._id}
        open={open}
        expanded={onExpand}
      >
        {children}
      </Collapsible>
    );
  }, []);
};
/* eslint-enable no-plusplus */

const buildHtml = (articles, articleTree, id, onArticleSelect, onExpand) => {
  let selectedArticlePath = [];
  if (id) {
    const selectedArticles = articles.filter(article => article._id === id);
    if (selectedArticles.length) {
      /* eslint-disable camelcase */
      const { topic, sub_topic, title } = selectedArticles[0];
      selectedArticlePath = [topic.name, title];
      if (sub_topic) {
        selectedArticlePath = selectedArticlePath.concat(
          sub_topic.name.split('>')
        );
      }
      /* eslint-enalbe camelcase */
    }
  }
  const articlesHtml = getChildren(
    articleTree,
    selectedArticlePath,
    onArticleSelect,
    onExpand
  );
  return articlesHtml;
};

export default buildHtml;
