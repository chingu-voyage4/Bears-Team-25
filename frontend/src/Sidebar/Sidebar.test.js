/* eslint-disable */
import React from 'react';
import { Link } from 'react-router-dom';

import Collapsible from './Collapsible';

import { getTree, getChildren } from './utils';

describe('creating tree', () => {
  let tree;
  const articles = [
    { _id: '1', topic: 'Voyage', sub_topic: 'About this wiki', title: 'Home' },
    {
      _id: '2',
      topic: 'Voyage',
      sub_topic: 'About this wiki',
      title: 'How to Contribute',
    },
    {
      _id: '3',
      topic: 'Voyage',
      sub_topic: 'About Voyages',
      title: 'About Voyages',
    },
  ];
  const expectedTree = {
    Voyage: {
      'About this wiki': { Home: '1', 'How to Contribute': '2' },
      'About Voyages': { 'About Voyages': '3' },
    },
  };
  const expectedElements = [
    <Collapsible key="Voyage" title="Voyage">
      <Collapsible key="About this wiki" title="About this wiki">
        <Link key="Home" to={'/cms/1'}>
          Home
        </Link>
        <Link key="How to Contribute" to={'/cms/2'}>
          How to Contribute
        </Link>
      </Collapsible>
      <Collapsible key="About Voyages" title="About Voyages">
        <Link key="About Voyages" to={'/cms/3'}>
          About Voyages
        </Link>
      </Collapsible>
    </Collapsible>,
  ];
  it('should create article tree', () => {
    tree = getTree(articles);
    expect(tree).toEqual(expectedTree);
  });
  it('should create elements', () => {
    const elements = getChildren(tree);
    expect(elements).toEqual(expectedElements);
  });
});
