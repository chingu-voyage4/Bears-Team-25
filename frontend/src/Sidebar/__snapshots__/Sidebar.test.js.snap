// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should initialise Sidebar correctly 1`] = `
Array [
  <Collapsible
    expanded={[Function]}
    id="1"
    open={false}
    title="Voyage"
>
    <Collapsible
        expanded={[Function]}
        id="10"
        open={false}
        title="About this Wiki"
    >
        <styled.div
            onClick={[Function]}
        >
            <Styled(NavLink)
                to="/cms/100"
            >
                Home
            </Styled(NavLink)>
        </styled.div>
        <styled.div
            onClick={[Function]}
        >
            <Styled(NavLink)
                to="/cms/101"
            >
                How to Contribute
            </Styled(NavLink)>
        </styled.div>
    </Collapsible>
    <Collapsible
        expanded={[Function]}
        id="11"
        open={false}
        title="About Voyages"
    >
        <styled.div
            onClick={[Function]}
        >
            <Styled(NavLink)
                to="/cms/102"
            >
                Voyage Roadmap
            </Styled(NavLink)>
        </styled.div>
    </Collapsible>
    <styled.div
        onClick={[Function]}
    >
        <Styled(NavLink)
            to="/cms/103"
        >
            Introduction
        </Styled(NavLink)>
    </styled.div>
</Collapsible>,
]
`;

exports[`should show mobile drawer 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Sidebar
    articles={
        Array [
            Object {
              "_id": "100",
              "order": 1,
              "sub_topic": Object {
                "_id": "10",
                "name": "About this Wiki",
                "order": 1,
                "parent": "1",
              },
              "title": "Home",
              "topic": Object {
                "_id": "1",
                "name": "Voyage",
                "order": 1,
              },
            },
            Object {
              "_id": "101",
              "order": 2,
              "sub_topic": Object {
                "_id": "10",
                "name": "About this Wiki",
                "order": 1,
                "parent": "1",
              },
              "title": "How to Contribute",
              "topic": Object {
                "_id": "1",
                "name": "Voyage",
                "order": 1,
              },
            },
            Object {
              "_id": "102",
              "order": 1,
              "sub_topic": Object {
                "_id": "11",
                "name": "About Voyages",
                "order": 2,
                "parent": "1",
              },
              "title": "Voyage Roadmap",
              "topic": Object {
                "_id": "1",
                "name": "Voyage",
                "order": 1,
              },
            },
            Object {
              "_id": "103",
              "order": 1,
              "sub_topic": null,
              "title": "Introduction",
              "topic": Object {
                "_id": "1",
                "name": "Voyage",
                "order": 1,
              },
            },
          ]
    }
    match={
        Object {
            "params": Object {
              "id": "",
            },
          }
    }
    onClick={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <WithStyles(IconButton)
          aria-label="Menu"
          colors="inherit"
          onClick={[Function]}
>
          <pure(Menu) />
</WithStyles(IconButton)>,
        <WithStyles(Drawer)
          onClose={[Function]}
          open={true}
>
          <styled.div
                    onClick={[Function]}
                    onKeyDown={[Function]}
                    role="button"
                    tabIndex={0}
          />
          <Collapsible
                    expanded={[Function]}
                    id="1"
                    open={false}
                    title="Voyage"
          >
                    <Collapsible
                              expanded={[Function]}
                              id="10"
                              open={false}
                              title="About this Wiki"
                    >
                              <styled.div
                                        onClick={[Function]}
                              >
                                        <Styled(NavLink)
                                                  to="/cms/100"
                                        >
                                                  Home
                                        </Styled(NavLink)>
                              </styled.div>
                              <styled.div
                                        onClick={[Function]}
                              >
                                        <Styled(NavLink)
                                                  to="/cms/101"
                                        >
                                                  How to Contribute
                                        </Styled(NavLink)>
                              </styled.div>
                    </Collapsible>
                    <Collapsible
                              expanded={[Function]}
                              id="11"
                              open={false}
                              title="About Voyages"
                    >
                              <styled.div
                                        onClick={[Function]}
                              >
                                        <Styled(NavLink)
                                                  to="/cms/102"
                                        >
                                                  Voyage Roadmap
                                        </Styled(NavLink)>
                              </styled.div>
                    </Collapsible>
                    <styled.div
                              onClick={[Function]}
                    >
                              <Styled(NavLink)
                                        to="/cms/103"
                              >
                                        Introduction
                              </Styled(NavLink)>
                    </styled.div>
          </Collapsible>
</WithStyles(Drawer)>,
      ],
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "aria-label": "Menu",
          "children": <pure(Menu) />,
          "colors": "inherit",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <styled.div
              onClick={[Function]}
              onKeyDown={[Function]}
              role="button"
              tabIndex={0}
/>,
            Array [
              <Collapsible
                expanded={[Function]}
                id="1"
                open={false}
                title="Voyage"
>
                <Collapsible
                                expanded={[Function]}
                                id="10"
                                open={false}
                                title="About this Wiki"
                >
                                <styled.div
                                                onClick={[Function]}
                                >
                                                <Styled(NavLink)
                                                                to="/cms/100"
                                                >
                                                                Home
                                                </Styled(NavLink)>
                                </styled.div>
                                <styled.div
                                                onClick={[Function]}
                                >
                                                <Styled(NavLink)
                                                                to="/cms/101"
                                                >
                                                                How to Contribute
                                                </Styled(NavLink)>
                                </styled.div>
                </Collapsible>
                <Collapsible
                                expanded={[Function]}
                                id="11"
                                open={false}
                                title="About Voyages"
                >
                                <styled.div
                                                onClick={[Function]}
                                >
                                                <Styled(NavLink)
                                                                to="/cms/102"
                                                >
                                                                Voyage Roadmap
                                                </Styled(NavLink)>
                                </styled.div>
                </Collapsible>
                <styled.div
                                onClick={[Function]}
                >
                                <Styled(NavLink)
                                                to="/cms/103"
                                >
                                                Introduction
                                </Styled(NavLink)>
                </styled.div>
</Collapsible>,
            ],
          ],
          "onClose": [Function],
          "open": true,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "onClick": [Function],
              "onKeyDown": [Function],
              "role": "button",
              "tabIndex": 0,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Collapsible
                  expanded={[Function]}
                  id="10"
                  open={false}
                  title="About this Wiki"
>
                  <styled.div
                                    onClick={[Function]}
                  >
                                    <Styled(NavLink)
                                                      to="/cms/100"
                                    >
                                                      Home
                                    </Styled(NavLink)>
                  </styled.div>
                  <styled.div
                                    onClick={[Function]}
                  >
                                    <Styled(NavLink)
                                                      to="/cms/101"
                                    >
                                                      How to Contribute
                                    </Styled(NavLink)>
                  </styled.div>
</Collapsible>,
                <Collapsible
                  expanded={[Function]}
                  id="11"
                  open={false}
                  title="About Voyages"
>
                  <styled.div
                                    onClick={[Function]}
                  >
                                    <Styled(NavLink)
                                                      to="/cms/102"
                                    >
                                                      Voyage Roadmap
                                    </Styled(NavLink)>
                  </styled.div>
</Collapsible>,
                <styled.div
                  onClick={[Function]}
>
                  <Styled(NavLink)
                                    to="/cms/103"
                  >
                                    Introduction
                  </Styled(NavLink)>
</styled.div>,
              ],
              "expanded": [Function],
              "id": "1",
              "open": false,
              "title": "Voyage",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "10",
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    <styled.div
                      onClick={[Function]}
>
                      <Styled(NavLink)
                                            to="/cms/100"
                      >
                                            Home
                      </Styled(NavLink)>
</styled.div>,
                    <styled.div
                      onClick={[Function]}
>
                      <Styled(NavLink)
                                            to="/cms/101"
                      >
                                            How to Contribute
                      </Styled(NavLink)>
</styled.div>,
                  ],
                  "expanded": [Function],
                  "id": "10",
                  "open": false,
                  "title": "About this Wiki",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": "100",
                    "nodeType": "class",
                    "props": Object {
                      "children": <Styled(NavLink)
                        to="/cms/100"
>
                        Home
</Styled(NavLink)>,
                      "onClick": [Function],
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "children": "Home",
                        "to": "/cms/100",
                      },
                      "ref": null,
                      "rendered": "Home",
                      "type": [Function],
                    },
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": "101",
                    "nodeType": "class",
                    "props": Object {
                      "children": <Styled(NavLink)
                        to="/cms/101"
>
                        How to Contribute
</Styled(NavLink)>,
                      "onClick": [Function],
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "children": "How to Contribute",
                        "to": "/cms/101",
                      },
                      "ref": null,
                      "rendered": "How to Contribute",
                      "type": [Function],
                    },
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": "11",
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    <styled.div
                      onClick={[Function]}
>
                      <Styled(NavLink)
                                            to="/cms/102"
                      >
                                            Voyage Roadmap
                      </Styled(NavLink)>
</styled.div>,
                  ],
                  "expanded": [Function],
                  "id": "11",
                  "open": false,
                  "title": "About Voyages",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": "102",
                    "nodeType": "class",
                    "props": Object {
                      "children": <Styled(NavLink)
                        to="/cms/102"
>
                        Voyage Roadmap
</Styled(NavLink)>,
                      "onClick": [Function],
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "children": "Voyage Roadmap",
                        "to": "/cms/102",
                      },
                      "ref": null,
                      "rendered": "Voyage Roadmap",
                      "type": [Function],
                    },
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": "103",
                "nodeType": "class",
                "props": Object {
                  "children": <Styled(NavLink)
                    to="/cms/103"
>
                    Introduction
</Styled(NavLink)>,
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Introduction",
                    "to": "/cms/103",
                  },
                  "ref": null,
                  "rendered": "Introduction",
                  "type": [Function],
                },
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <WithStyles(IconButton)
            aria-label="Menu"
            colors="inherit"
            onClick={[Function]}
>
            <pure(Menu) />
</WithStyles(IconButton)>,
          <WithStyles(Drawer)
            onClose={[Function]}
            open={true}
>
            <styled.div
                        onClick={[Function]}
                        onKeyDown={[Function]}
                        role="button"
                        tabIndex={0}
            />
            <Collapsible
                        expanded={[Function]}
                        id="1"
                        open={false}
                        title="Voyage"
            >
                        <Collapsible
                                    expanded={[Function]}
                                    id="10"
                                    open={false}
                                    title="About this Wiki"
                        >
                                    <styled.div
                                                onClick={[Function]}
                                    >
                                                <Styled(NavLink)
                                                            to="/cms/100"
                                                >
                                                            Home
                                                </Styled(NavLink)>
                                    </styled.div>
                                    <styled.div
                                                onClick={[Function]}
                                    >
                                                <Styled(NavLink)
                                                            to="/cms/101"
                                                >
                                                            How to Contribute
                                                </Styled(NavLink)>
                                    </styled.div>
                        </Collapsible>
                        <Collapsible
                                    expanded={[Function]}
                                    id="11"
                                    open={false}
                                    title="About Voyages"
                        >
                                    <styled.div
                                                onClick={[Function]}
                                    >
                                                <Styled(NavLink)
                                                            to="/cms/102"
                                                >
                                                            Voyage Roadmap
                                                </Styled(NavLink)>
                                    </styled.div>
                        </Collapsible>
                        <styled.div
                                    onClick={[Function]}
                        >
                                    <Styled(NavLink)
                                                to="/cms/103"
                                    >
                                                Introduction
                                    </Styled(NavLink)>
                        </styled.div>
            </Collapsible>
</WithStyles(Drawer)>,
        ],
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "aria-label": "Menu",
            "children": <pure(Menu) />,
            "colors": "inherit",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <styled.div
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={0}
/>,
              Array [
                <Collapsible
                  expanded={[Function]}
                  id="1"
                  open={false}
                  title="Voyage"
>
                  <Collapsible
                                    expanded={[Function]}
                                    id="10"
                                    open={false}
                                    title="About this Wiki"
                  >
                                    <styled.div
                                                      onClick={[Function]}
                                    >
                                                      <Styled(NavLink)
                                                                        to="/cms/100"
                                                      >
                                                                        Home
                                                      </Styled(NavLink)>
                                    </styled.div>
                                    <styled.div
                                                      onClick={[Function]}
                                    >
                                                      <Styled(NavLink)
                                                                        to="/cms/101"
                                                      >
                                                                        How to Contribute
                                                      </Styled(NavLink)>
                                    </styled.div>
                  </Collapsible>
                  <Collapsible
                                    expanded={[Function]}
                                    id="11"
                                    open={false}
                                    title="About Voyages"
                  >
                                    <styled.div
                                                      onClick={[Function]}
                                    >
                                                      <Styled(NavLink)
                                                                        to="/cms/102"
                                                      >
                                                                        Voyage Roadmap
                                                      </Styled(NavLink)>
                                    </styled.div>
                  </Collapsible>
                  <styled.div
                                    onClick={[Function]}
                  >
                                    <Styled(NavLink)
                                                      to="/cms/103"
                                    >
                                                      Introduction
                                    </Styled(NavLink)>
                  </styled.div>
</Collapsible>,
              ],
            ],
            "onClose": [Function],
            "open": true,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "onClick": [Function],
                "onKeyDown": [Function],
                "role": "button",
                "tabIndex": 0,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <Collapsible
                    expanded={[Function]}
                    id="10"
                    open={false}
                    title="About this Wiki"
>
                    <styled.div
                                        onClick={[Function]}
                    >
                                        <Styled(NavLink)
                                                            to="/cms/100"
                                        >
                                                            Home
                                        </Styled(NavLink)>
                    </styled.div>
                    <styled.div
                                        onClick={[Function]}
                    >
                                        <Styled(NavLink)
                                                            to="/cms/101"
                                        >
                                                            How to Contribute
                                        </Styled(NavLink)>
                    </styled.div>
</Collapsible>,
                  <Collapsible
                    expanded={[Function]}
                    id="11"
                    open={false}
                    title="About Voyages"
>
                    <styled.div
                                        onClick={[Function]}
                    >
                                        <Styled(NavLink)
                                                            to="/cms/102"
                                        >
                                                            Voyage Roadmap
                                        </Styled(NavLink)>
                    </styled.div>
</Collapsible>,
                  <styled.div
                    onClick={[Function]}
>
                    <Styled(NavLink)
                                        to="/cms/103"
                    >
                                        Introduction
                    </Styled(NavLink)>
</styled.div>,
                ],
                "expanded": [Function],
                "id": "1",
                "open": false,
                "title": "Voyage",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": "10",
                  "nodeType": "class",
                  "props": Object {
                    "children": Array [
                      <styled.div
                        onClick={[Function]}
>
                        <Styled(NavLink)
                                                to="/cms/100"
                        >
                                                Home
                        </Styled(NavLink)>
</styled.div>,
                      <styled.div
                        onClick={[Function]}
>
                        <Styled(NavLink)
                                                to="/cms/101"
                        >
                                                How to Contribute
                        </Styled(NavLink)>
</styled.div>,
                    ],
                    "expanded": [Function],
                    "id": "10",
                    "open": false,
                    "title": "About this Wiki",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": "100",
                      "nodeType": "class",
                      "props": Object {
                        "children": <Styled(NavLink)
                          to="/cms/100"
>
                          Home
</Styled(NavLink)>,
                        "onClick": [Function],
                      },
                      "ref": null,
                      "rendered": Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "class",
                        "props": Object {
                          "children": "Home",
                          "to": "/cms/100",
                        },
                        "ref": null,
                        "rendered": "Home",
                        "type": [Function],
                      },
                      "type": [Function],
                    },
                    Object {
                      "instance": null,
                      "key": "101",
                      "nodeType": "class",
                      "props": Object {
                        "children": <Styled(NavLink)
                          to="/cms/101"
>
                          How to Contribute
</Styled(NavLink)>,
                        "onClick": [Function],
                      },
                      "ref": null,
                      "rendered": Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "class",
                        "props": Object {
                          "children": "How to Contribute",
                          "to": "/cms/101",
                        },
                        "ref": null,
                        "rendered": "How to Contribute",
                        "type": [Function],
                      },
                      "type": [Function],
                    },
                  ],
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": "11",
                  "nodeType": "class",
                  "props": Object {
                    "children": Array [
                      <styled.div
                        onClick={[Function]}
>
                        <Styled(NavLink)
                                                to="/cms/102"
                        >
                                                Voyage Roadmap
                        </Styled(NavLink)>
</styled.div>,
                    ],
                    "expanded": [Function],
                    "id": "11",
                    "open": false,
                    "title": "About Voyages",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": "102",
                      "nodeType": "class",
                      "props": Object {
                        "children": <Styled(NavLink)
                          to="/cms/102"
>
                          Voyage Roadmap
</Styled(NavLink)>,
                        "onClick": [Function],
                      },
                      "ref": null,
                      "rendered": Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "class",
                        "props": Object {
                          "children": "Voyage Roadmap",
                          "to": "/cms/102",
                        },
                        "ref": null,
                        "rendered": "Voyage Roadmap",
                        "type": [Function],
                      },
                      "type": [Function],
                    },
                  ],
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": "103",
                  "nodeType": "class",
                  "props": Object {
                    "children": <Styled(NavLink)
                      to="/cms/103"
>
                      Introduction
</Styled(NavLink)>,
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "Introduction",
                      "to": "/cms/103",
                    },
                    "ref": null,
                    "rendered": "Introduction",
                    "type": [Function],
                  },
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
